app/models/measure_collection.rb

```diff
@@ -1,90 +1,56 @@
 class MeasureCollection
   include Enumerable
 
-  attr_accessor :measures, :presenter_klass
+  attr_accessor :measures
 
+  delegate :present?, to: :measures
   delegate :size, :length, :collect, :map, :each, :all?, :include?, :to_ary, to: :to_a
 
-  def initialize(measures, presenter_klass = MeasurePresenter)
+  def initialize(measures)
     @measures = measures.clone
-    @presenter_klass = presenter_klass
   end
 
   def for_country(country)
-    self.measures = measures.select { |m| m.relevant_for_country?(country) }
-    self
+    self.class.new(measures.select { |m| m.relevant_for_country?(country) })
   end
 
   def vat_excise
-    @vat_excise ||= begin
-      self.measures = measures.select(&:vat_excise?)
-      self
-    end
+    self.class.new(measures.select(&:vat_excise?))
   end
 
   def import_controls
-    @import_controls ||= begin
-      self.measures = measures.select(&:import_controls?)
-      self
-    end
+    self.class.new(measures.select(&:import_controls?))
   end
 
   def trade_remedies
-    @trade_remedies ||= begin
-      self.measures = measures.select(&:trade_remedies?)
-      self
-    end
+    self.class.new(measures.select(&:trade_remedies?))
   end
 
   def customs_duties
-    @customs_duties ||= begin
-      self.measures = measures.select(&:customs_duties?)
-      self
-    end
+    self.class.new(measures.select(&:customs_duties?))
   end
 
   def quotas
-    @quotas ||= begin
-      self.measures = measures.select(&:quotas?)
-      self
-    end
+    self.class.new(measures.select(&:quotas?))
   end
 
   def vat
-    self.measures = measures.select(&:vat?)
-    self
+    self.class.new(measures.select(&:vat?))
   end
 
   def national
-    self.measures = measures.select(&:national?)
-    self
+    self.class.new(measures.select(&:national?))
   end
 
   def third_country_duty
-    self.measures = measures.select(&:third_country?)
-    self
+    self.class.new(measures.select(&:third_country?))
   end
 
   def supplementary
-    self.measures = measures.select(&:supplementary?)
-    self
+    self.class.new(measures.select(&:supplementary?))
   end
 
   def to_a
-    if presenter_klass.present?
-      present_with(presenter_klass)
-    else
-      measures
-    end
-  end
-
-  def present?
-    measures.any?
-  end
-
-  private
-
-  def present_with(presenter_klass)
-    measures.map { |entry| presenter_klass.new(entry) }
+    measures.map { |entry| MeasurePresenter.new(entry) }
```

Comment:
I couldn't see any situation in the code where we were **not** using the MeasurePresenter class when interacting with the Measure model via a collection. AFAIK we look at measures on a declarable via:

1. declarable#overview_measures
2. declarable#import_measures
3. declarable#export_measures

Reply:
that's reasonable, I should work